als-server/shared/src/test/scala/org/mulesoft/als/server/modules/diagnostic/support/avro/AvroCleanDiagnosticTest.scala

```diff
@@ -0,0 +1,67 @@
+package org.mulesoft.als.server.modules.diagnostic.support.avro
+
+import amf.core.client.common.validation.ProfileNames
+import org.mulesoft.als.server.modules.diagnostic.BasicCleanDiagnosticTest
+import org.mulesoft.lsp.feature.diagnostic.DiagnosticSeverity
+
+
+class AvroCleanDiagnosticTest extends BasicCleanDiagnosticTest {
+
+  test("Async importing valid Avro") {
+    withServer(buildServer()) { server =>
+      for {
+        d <- requestCleanDiagnostic(server)(filePath("avro/async26-imports-valid-avro.yaml"))
+      } yield {
+        server.shutdown()
+        d.foreach(filediag => assert(filediag.diagnostics.isEmpty))
+        assert(d.size == 2)
+        assert(d.head.profile == ProfileNames.ASYNC26)
+      }
+    }
+  }
+
+  test("Async importing invalid Avro") {
+    withServer(buildServer()) { server =>
+      for {
+        d <- requestCleanDiagnostic(server)(filePath("avro/async26-imports-invalid-avro.yaml"))
+      } yield {
+        server.shutdown()
+        assert(d.size == 2)
+        val avroFileDiagnostics = d.find(d => d.uri.contains("avro/async26-imports-invalid-avro.yaml"))
+        assert(avroFileDiagnostics.isDefined)
+        assert(avroFileDiagnostics.get.diagnostics.length == 1)
+        val errorMessage = avroFileDiagnostics.get.diagnostics.head.message.toLowerCase()
```

Comment:
maybe instead of making a `.get` anda a `head` do a `find` or `collect` so we do not depend on the order of the set and instead look for the specific uri which should have the file (and in case there is an error it would be a cleaner message than a `None .get` or something of the sorts)